nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Utilities/MSBuildHelperTests.cs

```diff
@@ -319,6 +319,115 @@ public async Task AllPackageDependencies_DoNotIncludeUpdateOnlyPackages()
         Assert.Equal(expectedDependencies, actualDependencies);
     }
 
+    [Fact]
+    public async Task GetAllPackageDependencies_NugetConfigInvalid_DoesNotThrow()
+    {
+        var nugetPackagesDirectory = Environment.GetEnvironmentVariable("NUGET_PACKAGES");
+        var nugetHttpCacheDirectory = Environment.GetEnvironmentVariable("NUGET_HTTP_CACHE_PATH");
+
+        try
+        {
+            using var temp = new TemporaryDirectory();
+
+            // It is important to have empty NuGet caches for this test, so override them with temp directories.
+            var tempNuGetPackagesDirectory = Path.Combine(temp.DirectoryPath, ".nuget", "packages");
+            Environment.SetEnvironmentVariable("NUGET_PACKAGES", tempNuGetPackagesDirectory);
```

Comment:
With us using this pattern more. Wonder if we could encapsulate this in an IDisposable?

Reply:
I was thinking about doing an `IDisposable`, but it gets a bit weird in this case because we only actually care about restoring some of the Environment variables. I think given that I was supposed to finish this last week I'll hold off on doing that until I'm back in the area.